umad-02-keys-and-authentication.md

```diff
@@ -25,42 +25,50 @@ section.
 
 ## Public Key Exchange
 
-VASPs expose their public keys to other VASPs by responding to `GET` requests at the endpoint
+The UMA protocol relies on X.509 certificates for public key exchange among VASPs. If a VASP trusts the party that they
+wish to transact with, typically established through prior communication, self-signed certificates are a suitable
+solution for key exchange. VASPs can create self-signed certificates wrapping the public keys generated above using
+common tools such as `openssl`, and share these certificates with their counterparty. It is important to note that
+self-signed certificates don't provide a good revocation mechanism, so it is recommended to use a short caching
+duration (on the order of a few minutes) to minimize the risk of key compromise.
+
+In cases where VASPs don't recognize their counterparty's identity, it is recommended to use signed certificates
+issued by a trusted source, such as a VASP ID Authority or a Certificate Authority. When a VASP receives a signed
```

Comment:
nit: I'd remove all references to "Certificate Authority" here. That has another specific meaning in TLS world, so I don't want to confuse things and let people think they can reuse ssl certs. Let's define "VASP ID Authority" first and then use it throughout.